app/src/main/java/com/teamagam/gimelgimel/app/common/utils/BiMap.java

```diff
@@ -0,0 +1,52 @@
+package com.teamagam.gimelgimel.app.common.utils;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+public class BiMap<TKey, TValue> {
+
+    private final Map<TKey, TValue> mMap;
+    private final Map<TValue, TKey> mInverseMap;
+
+    public BiMap() {
+        mMap = new TreeMap<>();
+        mInverseMap = new TreeMap<>();
+    }
+
+    public void put(TKey key, TValue value) {
+        mMap.put(key, value);
+        mInverseMap.put(value, key);
+    }
+
+    public TValue getValue(TKey key) {
+        return mMap.get(key);
+    }
+
+    public TKey getKey(TValue value) {
+        return mInverseMap.get(value);
+    }
+
+    public boolean containsKey(TKey key) {
+        return mMap.containsKey(key);
+    }
+
+    public boolean containsValue(TValue value) {
+        return mInverseMap.containsKey(value);
+    }
+
+    public void removeByKey(TKey key) {
+        TValue value = mMap.get(key);
+        mMap.remove(key);
+        mInverseMap.remove(value);
+    }
+
+    public void removeByValue(TValue value) {
+        TKey key = mInverseMap.get(value);
+        mInverseMap.remove(value);
+        mMap.remove(key);
+    }
+
+    public int size() {
+        return mMap.size();
+    }
+}
```

Comment:
I think key and value are good choice here mainly because of their contribution to naming the getByX and removeByX methods.
The uniqueness is support and exists, but it's not enforced.
Do you think I should enforce it ?